### ISSUE TITLE
Unable to join world - String too Long

### ISSUE THREAD

**Issue by WinterPhoenix-602** [Abbie5/emi-ores#24]
It seems that something in this addon is causing me to be unable to join my singleplayer worlds. I can create them just fine, and it doesn't actually crash, it just immediately kicks me and says my player data is invalid. Here's the error. The entire log is included as well. Is this just because I have too many mods that add different ores? I'm playing in 1.20.1.
```
io.netty.handler.codec.EncoderException: String too big (was 56689 characters, max 32767)
	at net.minecraft.class_2540.method_10788(class_2540.java:1876) ~[client-intermediary.jar:?]
	at net.minecraft.class_2540.method_10814(class_2540.java:701) ~[client-intermediary.jar:?]
	at net.minecraft.class_2540.method_49395(class_2540.java:159) ~[client-intermediary.jar:?]
	at cc.abbie.emi_ores.networking.packet.S2CSendFeaturesPacket.lambda$write$1(S2CSendFeaturesPacket.java:37) ~[emi_ores-1.0+1.20.1+fabric.jar:?]
	at net.minecraft.class_2540.method_34064(class_2540.java:267) ~[client-intermediary.jar:?]
	at java.util.HashMap.forEach(HashMap.java:1421) ~[?:?]
	at net.minecraft.class_2540.method_34063(class_2540.java:265) ~[client-intermediary.jar:?]
	at cc.abbie.emi_ores.networking.packet.S2CSendFeaturesPacket.write(S2CSendFeaturesPacket.java:34) ~[emi_ores-1.0+1.20.1+fabric.jar:?]
	at cc.abbie.emi_ores.fabric.EmiOresFabric.lambda$onSyncDatapackContents$1(EmiOresFabric.java:28) ~[emi_ores-1.0+1.20.1+fabric.jar:?]
	at cc.abbie.emi_ores.networking.FeaturesSender.onSyncDataPackContents(FeaturesSender.java:65) ~[emi_ores-1.0+1.20.1+fabric.jar:?]
	at cc.abbie.emi_ores.fabric.EmiOresFabric.onSyncDatapackContents(EmiOresFabric.java:22) ~[emi_ores-1.0+1.20.1+fabric.jar:?]
	at net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents.lambda$static$8(ServerLifecycleEvents.java:86) ~[fabric-lifecycle-events-v1-2.2.2+1802ada577-15099849194dcb7e.jar:?]
	at net.minecraft.class_3324.handler$dmc000$fabric-lifecycle-events-v1$hookOnPlayerConnect(class_3324.java:8021) ~[client-intermediary.jar:?]
	at net.minecraft.class_3324.method_14570(class_3324.java:219) ~[client-intermediary.jar:?]
	at net.minecraft.class_3248.method_33800(class_3248.java:131) ~[client-intermediary.jar:?]
	at net.minecraft.class_3248.method_14384(class_3248.java:118) ~[client-intermediary.jar:?]
	at net.minecraft.class_3248.redirect$dnk000$fabric-networking-api-v1$handlePlayerJoin(class_3248.java:563) ~[client-intermediary.jar:?]
	at net.minecraft.class_3248.method_18784(class_3248.java:68) ~[client-intermediary.jar:?]
	at net.minecraft.class_2535.method_10754(class_2535.java:259) ~[client-intermediary.jar:?]
	at net.minecraft.class_3242.method_14357(class_3242.java:172) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:908) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:824) ~[client-intermediary.jar:?]
	at net.minecraft.class_1132.method_3748(class_1132.java:105) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:671) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265) ~[client-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[13:46:08] [Server thread/INFO]: WinterPhoenix lost connection: Invalid player data
[13:46:08] [Server thread/INFO]: WinterPhoenix left the game
[13:46:08] [Server thread/INFO]: remove player wand
[13:46:08] [Server thread/INFO]: Stopping singleplayer server as player logged out
[13:46:08] [Server thread/INFO]: Stopping server
[13:46:08] [Server thread/INFO]: Saving players
[13:46:08] [Server thread/INFO]: Saving worlds
```
[emiOresLog.txt](https://github.com/user-attachments/files/17385044/emiOresLog.txt)

**Comment by Abbie5** [comment]
Hi, can you please test again with this testing build? It uses a different method of encoding the features that should hopefully work with larger sizes.
[emi_ores-1.0-TESTING+1.20.1+fabric.jar.zip](https://github.com/user-attachments/files/17386819/emi_ores-1.0-TESTING%2B1.20.1%2Bfabric.jar.zip)

**Comment by WinterPhoenix-602** [comment]
That version seems to work, thanks! Glad you were already working on something that would fix this, it's very helpful when I have so many mods installed. It does seem like EMI takes a while to load after first joining the world, but that's just a few seconds delay.

**Comment by Abbie5** [comment]
fixed in 81b0b98
